iroh-net/src/discovery/local_node_discovery.rs

```diff
@@ -0,0 +1,250 @@
+use std::{
+    collections::{BTreeSet, HashMap},
+    net::{IpAddr, SocketAddr},
+    time::Duration,
+};
+
+use anyhow::Result;
+use derive_more::FromStr;
+use futures_lite::{stream::Boxed as BoxStream, StreamExt};
+
+use flume::Sender;
+use iroh_base::key::PublicKey;
+use swarm_discovery::{Discoverer, DropGuard, Peer};
+use tokio::task::{JoinHandle, JoinSet};
+
+use crate::{
+    discovery::{Discovery, DiscoveryItem},
+    AddrInfo, Endpoint, NodeId,
+};
+
+/// The n0 local node discovery name
+// TODO(ramfox): bikeshed
+const N0_MDNS_SWARM: &str = "iroh.local.node.discovery";
```

Comment:
Since this is used many times within each mDNS packet and size is limited, I’d recommend to use the shortest name you can think of.

I half-read about a compression feature in the DNS spec that would help, but at least I haven’t done anything specifically to implement it—will have to look into hickory to find out.